library/Vanilla/Models/ThemeModel.php

```diff
@@ -233,4 +250,24 @@ public function deleteAsset(string $themeKey, string $assetKey) {
         $provider = $this->getThemeProvider($themeKey);
         return $provider->deleteAsset($themeKey, $assetKey);
     }
+
+    /**
+     * Generate a theme preview from the variables.
+     *
+     * @param array $theme
+     * @return array
+     */
+    public function generateThemePreview(array $theme): array {
+        $variables = $theme["assets"]["variables"]->getData('array');
+        $preview = [];
+        if ($variables) {
+            $preview['global.mainColors.primary'] = $variables['global']['mainColors']['primary'] ?? null;
+            $preview['global.mainColors.bg'] = $variables['global']['mainColors']['bg'] ?? null;
+            $preview['global.mainColors.fg'] = $variables['global']['mainColors']['fg'] ?? null;
+            $preview['titleBar.colors.bg'] = $variables['titleBar']['colors']['bg'] ?? null;
+            $preview['titleBar.colors.fg'] = $variables['titleBar']['colors']['fg'] ?? null;
+            $preview['splash.outerBackground.image'] = $variables['splash']['outerBackground']['image'] ?? null;
+        }
+        return $preview;
+    }
```

Comment:
This is failing on my machine with the `foundation` theme. For some reason it comes up with no variables asset.

![image](https://user-images.githubusercontent.com/1770056/71196269-b740c680-225d-11ea-91ce-7d1d6b385232.png)